exec/java-exec/src/test/java/org/apache/drill/exec/store/bson/TestBsonRecordReader.java

```diff
@@ -49,17 +50,20 @@
 import org.junit.BeforeClass;
 import org.junit.Test;
 
-public class TestBsonRecordReader extends BaseTestQuery {
+public class TestBsonRecordReader {
+  private static BufferAllocator allocator;
   private static VectorContainerWriter writer;
   private static TestOutputMutator mutator;
+  private static DrillBuf buffer;
   private static BsonRecordReader bsonReader;
 
   @BeforeClass
   public static void setUp() {
-    BufferAllocator bufferAllocator = getDrillbitContext().getAllocator();
-    mutator = new TestOutputMutator(bufferAllocator);
+    allocator = new RootAllocator(9_000_00);
```

Comment:
It does not sound right as there is not much data used in the test.  I'd suggest filing a JIRA to look into why it takes 900 KB or almost 1 MB to hold data and using unlimited allocator (the same as before).